rsocket/framing/FramedReader.cpp

```diff
@@ -12,214 +12,206 @@ namespace rsocket {
-    parseFrames();
-  }
+void FramedReader::onNext(std::unique_ptr<folly::IOBuf> payload) {
+  VLOG(4) << "incoming bytes length=" << payload->length() << '\n'
+          << hexDump(payload->clone()->moveToFbString());
+  payloadQueue_.append(std::move(payload));
+  parseFrames();
 }
 
 void FramedReader::parseFrames() {
   if (dispatchingFrames_) {
     return;
   }
 
-  // delivering onNext can trigger termination and destroying this instance
-  // while in this method so we will make sure we survive
+  // Delivering onNext can trigger termination and destroy this instance.
   auto thisPtr = this->ref_from_this(this);
 
   dispatchingFrames_ = true;
 
-  while (allowance_.canAcquire() && frames_) {
+  while (allowance_.canAcquire() && inner_) {
     if (!ensureOrAutodetectProtocolVersion()) {
-      // at this point we dont have enough bytes on the wire
-      // or we errored out
+      // At this point we dont have enough bytes on the wire or we errored out.
       break;
     }
 
-    if (payloadQueue_.chainLength() < getFrameSizeFieldLength()) {
-      // we don't even have the next frame size value
+    auto const frameSizeFieldLen = frameSizeFieldLength(*version_);
+    if (payloadQueue_.chainLength() < frameSizeFieldLen) {
+      // We don't even have the next frame size value.
       break;
     }
 
-    const auto nextFrameSize = readFrameLength();
-
-    // so if the size value is less than minimal frame length something is wrong
-    if (nextFrameSize < getFrameMinimalLength()) {
-      error("invalid data stream");
+    auto const nextFrameSize = readFrameLength();
+    if (nextFrameSize < minimalFrameLength(*version_)) {
+      error("Invalid frame - Frame size smaller than minimum");
       break;
     }
 
     if (payloadQueue_.chainLength() <
-        getFrameSizeWithLengthField(nextFrameSize)) {
-      // need to accumulate more data
+        frameSizeWithLengthField(*version_, nextFrameSize)) {
+      // Need to accumulate more data.
       break;
     }
-    payloadQueue_.trimStart(getFrameSizeFieldLength());
-    auto payloadSize = getPayloadSize(nextFrameSize);
-    // IOBufQueue::split(0) returns a null unique_ptr, so we create an empty
-    // IOBuf object and pass a unique_ptr to it instead. This simplifies
-    // clients' code because they can assume the pointer is non-null.
-    auto nextFrame = payloadSize != 0 ? payloadQueue_.split(payloadSize)
-                                      : folly::IOBuf::create(0);
+
+    payloadQueue_.trimStart(frameSizeFieldLen);
+    auto payloadSize = frameSizeWithoutLengthField(*version_, nextFrameSize);
+
+    DCHECK_GT(payloadSize, 0)
+        << "folly::IOBufQueue::split(0) returns a nullptr, can't have that";
+    auto nextFrame = payloadQueue_.split(payloadSize);
 
     CHECK(allowance_.tryAcquire(1));
 
-    VLOG(4) << "parsed frame length=" << nextFrame->length() << std::endl
+    VLOG(4) << "parsed frame length=" << nextFrame->length() << '\n'
```

Comment:
'\n' is not going to work on all platforms. On windows new line is encoded with 2 characters so the std:endl is internally inserting "\n\r"..

Reply:
Are we even compiling on Windows?  Also `std::endl` is the worst because it's defined to flush the stream.  This isn't the end of the world when it's being used in a `VLOG`, but still...